modules/programs/zsh.nix

```diff
@@ -322,7 +322,7 @@ in
     })
 
     (mkIf (cfg.dotDir != null) {
-      programs.zsh.sessionVariables.ZDOTDIR = zdotdir;
+      programs.zsh.sessionVariables.ZDOTDIR = if config.home.sessionVariableSetter == "pam" then "@{HOME}/" + cfg.dotDir else zdotdir;
```

Comment:
Hmm, how does `ZDOTDIR` end up in the `config.home.sessionVariables` variable? I don't see how it could happen…

In any case, when portability between pam and shell setters is needed then I'd suggest using `config.home.homeDirectory` which will expand to the absolute path to the user's home directory. E.g.,
```
zdotdir = config.home.homeDirectory + "/" + cfg.dotDir;
```